plugins/block-dynamic-connection/src/dynamic_if.js

```diff
@@ -0,0 +1,203 @@
+      const inputNumbers = inputs.split(',');
+      if (this.getInput('IF0')) {
+        this.removeInput('IF0');
+      }
+      if (this.getInput('DO0')) {
+        this.removeInput('DO0');
+      }
+      const first = inputNumbers[0];
+      this.appendValueInput('IF' + first)
+          .setCheck('Boolean')
+          .appendField(Blockly.Msg['CONTROLS_IF_MSG_IF'], 'if');
+      this.appendStatementInput('DO' + first)
+          .appendField(Blockly.Msg['CONTROLS_IF_MSG_THEN']);
+
+      for (let i = 1; i < inputNumbers.length; i++) {
+        this.appendValueInput('IF' + inputNumbers[i])
+            .setCheck('Boolean')
+            .appendField(Blockly.Msg['CONTROLS_IF_MSG_ELSEIF'], 'elseif');
+        this.appendStatementInput('DO' + inputNumbers[i])
+            .appendField(Blockly.Msg['CONTROLS_IF_MSG_THEN']);
+      }
+    }
+    const hasElse = xmlElement.getAttribute('else');
+    if (hasElse == 'true') {
+      this.appendStatementInput('ELSE')
+          .appendField(Blockly.Msg['CONTROLS_IF_MSG_ELSE'], 'else');
+    }
+    const next = parseInt(xmlElement.getAttribute('next'));
+    this.inputCounter = next;
+  },
+
+  /**
+   * Finds the index of a connection. Used to determine where in the block to
+   * insert new inputs.
+   * @param {!Blockly.Connection} connection A connection on this block.
+   * @return {number} The index of the connection in the this.inputList.
+   */
+  findInputIndexForConnection: function(connection) {
+    for (let i = 0; i < this.inputList.length; i++) {
+      const input = this.inputList[i];
+      if (input.connection == connection) {
+        return i;
+      }
+    }
+  },
+
+  /**
+   * Inserts a boolean value input and statement input at the specified index.
+   * @param {number} index Index of the input before which to add new inputs.
+   */
+  insertElseIf: function(index) {
+    const caseNumber = this.inputCounter;
+    this
+        .appendValueInput('IF' + caseNumber)
+        .setCheck('Boolean')
+        .appendField(Blockly.Msg['CONTROLS_IF_MSG_ELSEIF'], 'elseif');
+    this.appendStatementInput('DO' + caseNumber)
+        .appendField(Blockly.Msg['CONTROLS_IF_MSG_THEN']);
+    this.moveInputBefore('IF' + caseNumber, this.inputList[index].name);
+    this.moveInputBefore('DO' + caseNumber, this.inputList[index + 1].name);
+    this.inputCounter++;
+  },
+
+  /**
+   * Called when a block is dragged over one of the connections on this block.
+   * @param {!Blockly.Connection} connection The connection on this block that
+   * has a pending connection.
+   */
+  onPendingConnection: function(connection) {
+    if (connection.type === Blockly.NEXT_STATEMENT && !this.getInput('ELSE')) {
+      this.appendStatementInput('ELSE')
+          .appendField(Blockly.Msg['CONTROLS_IF_MSG_ELSE'], 'else');
+    }
+    const inputIndex = this.findInputIndexForConnection(connection);
+    const input = this.inputList[inputIndex];
+    if (connection.targetConnection && input.name.includes('IF')) {
```

Comment:
I was playing with the plugin, and ran into the problem where `input` was null here. I am having difficulty reproducing the error, however I think it makes sense to handle the case where `inputIndex` is `undefined`. Or, I think it would be better if we returned `null` from `firstInputIndexForConnection` and update the type to be `{?number}`. (The ? means that the type can be null.)